### PR TITLE
add release-monitoring.org fallback lookup method

### PR DESCRIPTION
**Pull request by stevendanna**
The release-monitoring.org API provides version information for open
source packages. While this information can be out of date or
incorrect, it appears reliable enough to use as a
method-of-last-resort.

This change adds a release-monitoring.org fallback if no other version
lookup methods work. The lookup is done via a static list of project
IDs that I've assembled.

Signed-off-by: Steven Danna <steve@chef.io>

### CODE REVIEW

**Comment by stevendanna** [comment]
Apologies for the delay on this, I went on vacation and then lost track of it for a bit.

My shell style is usually different from the rest of the script so I imagine I made a number of style mistakes, just let me know and I can fix them up.  I'm not too attached to the code here, the real work was looking up all the project IDs.

File: groundskeeper [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (stevendanna) [inline groundskeeper:1]: Not sure how you feel about this.  It makes things a bit easier on macOS which ships ancient bash by default.

File: groundskeeper [hunk 2]
```diff
@@ -10,6 +10,10 @@ STUDIO_TYPE="groundskeeper"
 DEBUG=${DEBUG:-0}
 ALL_PLANS=()
 SKIP_CLONE=${SKIP_CLONE:-0}
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd )"
+RELEASE_MONITORING_ID_FILE="${SCRIPT_DIR}/release_monitoring_ids"
+
+declare -A release_monitoring_ids
```
Comment (stevendanna) [inline groundskeeper:16]: The use of associative arrays means we now require Bash 4.0+

File: groundskeeper [hunk 3]
```diff
@@ -579,13 +653,12 @@ for plan in "${ALL_PLANS[@]}"; do
...
   plan_count_total=$((plan_count_total + 1))
 
-  if [ ${DEBUG} -eq 1 ]; then
+  if [ "${DEBUG}" -ge 1 ]; then
     echo
   fi
 
   # pv = Plan Version
   pv=$(plan_version "${plan}")
-  pv_status=$?
```
Comment (stevendanna) [inline groundskeeper:588]: This variable was unused.

**Comment by stevendanna** [comment]
Before:
```
[ Summary ] ------------------------------------------
  Total plans:     627
  Found versions:  244
```

After:
```
[ Summary ] ------------------------------------------
  Total plans:     627
  Found versions:  458
```

File: groundskeeper (hunk 2 above)
Comment (predominant) [reply groundskeeper:16]: I'm OK with that :)

File: groundskeeper (hunk 1 above)
Comment (predominant) [reply groundskeeper:1]: 👍

**Comment by predominant** [comment]
Absolutely amazing work!
tenor-220374684

**Pull request merged**